sources/models/src/modeled_types/credential_providers.rs

```diff
@@ -0,0 +1,128 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+// Just need serde's Error in scope to get its trait methods
+use super::error;
+use serde::de::Error as _;
+use snafu::ensure;
+use std::borrow::Borrow;
+use std::convert::TryFrom;
+use std::fmt;
+use std::ops::Deref;
+
+use crate::SingleLineString;
+
+/// CacheDurationValue is a string that represents the duration the plugin will
+/// cache credentials in-memory if a cache duration is not provided in the
+/// plugin response.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct CacheDurationValue {
+    inner: String,
+}
+
+lazy_static! {
+    // A duration string is a sequence of decimal numbers, each with a
+    // unit suffix, such as "300ms" or "2h45m".
+    // Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+    // Based on restraints from Go's `ParseDuration` funciton.
+    pub(crate) static ref DURATION_FORMAT: Regex = Regex::new(r"^([0-9]*[a-z]+)+$").unwrap();
+}
+
+impl TryFrom<&str> for CacheDurationValue {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, error::Error> {
+        ensure!(
+            DURATION_FORMAT.is_match(input),
+            error::InvalidCacheDurationValueSnafu { input }
+        );
+        Ok(CacheDurationValue {
+            inner: input.to_string(),
+        })
+    }
+}
+
+string_impls_for!(CacheDurationValue, "CacheDurationValue");
+
+#[cfg(test)]
+mod test_cache_duration_value {
+    use super::CacheDurationValue;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_cache_duration_value() {
+        for ok in &["12h", "120m", "2h15m"] {
+            CacheDurationValue::try_from(*ok).unwrap();
+        }
+    }
+
+    #[test]
+    fn invalid_cache_duration_value() {
+        for err in &["12 days", "💝", "Ï", ""] {
```

Comment:
If tightening the validation based on my comment on the regex, then we would want to add something like:

```suggestion
        for err in &["12 days", "💝", "Ï", "", "1minute"] {
```